rest_framework/validators.py

```diff
@@ -246,11 +235,10 @@ def filter_queryset(self, attrs, queryset, field_name, date_field_name):
         value = attrs[self.field]
         date = attrs[self.date_field]
 
-        filter_kwargs = {}
-        filter_kwargs[field_name] = value
-        filter_kwargs['%s__day' % date_field_name] = date.day
-        filter_kwargs['%s__month' % date_field_name] = date.month
-        filter_kwargs['%s__year' % date_field_name] = date.year
+        filter_kwargs = {field_name: value,
+                         f'{date_field_name}__day': date.day,
+                         f'{date_field_name}__month': date.month,
+                         f'{date_field_name}__year': date.year}
```

Comment:
This looks much nicer after your changes, doesn't it?

Could do with a bit of whitespace formatting too I guess...

```suggestion
        filter_kwargs = {
            field_name: value,
            f'{date_field_name}__day': date.day,
            f'{date_field_name}__month': date.month,
            f'{date_field_name}__year': date.year
        }
```

(Aside: I *thought* we checked for `black` styling automatically as part of our CI, but that doesn't seem to be the case.)

Reply:
I formatted it like that but I left the first line after the "{" but then i clicked that format button for flake8 and it re-formatted it this way and I wasn't going to fight it)  I do like this way better.